### PR TITLE
KAFKA-8227 DOCS Fixed missing links duality of streams tables

### PR DESCRIPTION
**Pull request by londoncalling**
## Description

- Jira ticket: https://issues.apache.org/jira/browse/KAFKA-8227
- Related PR: https://github.com/apache/kafka-site/pull/204
- DOCS update to fix missing and broken links
 
## Review Requests
@bbejeck , @mjsax , @JimGalasyn , @joel-hamill 

Signed-off-by: Victoria Bialas <vicky@confluent.io>

### CODE REVIEW

File: docs/streams/core-concepts.html
Comment (bbejeck) [inline docs/streams/core-concepts.html]: All the links point to the correct sections but I think we can get rid of all the `<code class="interpreted-text" data-role="ref">&lt;XX&gt;</code>` blocks in this section and elsewhere.  

I could be wrong, but It looked to me the `interpreted-text` sections were placeholders for setting the correct links pointing to other sections in the docs.

**Review by bbejeck — COMMENTED** [review]
@londoncalling thanks for the PR.  I just have some minor comments, but otherwise this PR LGTM.

**Comment by londoncalling** [comment]
@mjsax , @joel-hamill , @JimGalasyn this should be good to go after build completes, can someone merge for me who has permissions?

**Review by JimGalasyn — APPROVED** [review]

**Review by bbejeck — APPROVED** [review]
Thanks for the cleanup @londoncalling! LGTM

**Comment by bbejeck** [comment]
Merged #6625 into trunk

**Comment by bbejeck** [comment]
Thanks for fixing the links @londoncalling!

**Comment by bbejeck** [comment]
cherry-picked to 2.2, 2.1, and 2.0

**Pull request merged**
